oligo_designer_toolsuite/oligo_specificity_filter/_filter_bowtie.py

```diff
@@ -0,0 +1,337 @@
+            )
+        else:
+            self.min_mismatches = min_mismatches
+
+        if mismatch_region < 5:
+            raise ValueError(
+                "Choice of mismatch_region out of range for bowtie allignment tool. Please choose a value no less than 5"
+            )
+        else:
+            self.mismatch_region = mismatch_region
+
+        self.dir_bowtie = os.path.join(self.dir_output, "bowtie")
+        Path(self.dir_bowtie).mkdir(parents=True, exist_ok=True)
+
+        self.dir_probes = os.path.join(self.dir_output, "probes_bowtie")
+        Path(self.dir_probes).mkdir(parents=True, exist_ok=True)
+
+        self.file_removed_genes = os.path.join(
+            self.dir_probes, "genes_with_insufficient_probes.txt"
+        )
+
+    def run_bowtie(self):
+        """Run Bowtie alignment tool to find regions of local similarity between sequences, where sequences are probes and transcripts.
+        Bowtie identifies all allignments between the probes and transcripts and returns the number of mismatches and mismatch position for each alignment.
+
+        :return: DataFrame with processed bowtie alignment search results.
+        :rtype: pandas.DataFrame
+        """
+
+        def _run_bowtie_batch(batch_id):
+            """Run Bowtie alignment tool for all probes of one batch
+            :param batch_id: Batch ID.
+            :type batch_id: int
+            """
+
+            for subbatch_id in range(self.number_subbatches):
+                file_probe_fasta_batch = os.path.join(
+                    self.dir_annotations,
+                    "probes_sequence_batch{}_{}.fna".format(batch_id, subbatch_id),
+                )
+                file_bowtie_batch = os.path.join(
+                    self.dir_bowtie,
+                    "bowtie_batch{}_{}.txt".format(batch_id, subbatch_id),
+                )
+
+                if self.mismatch_region is not None:
+                    command = (
+                        "bowtie transcriptome -f -a -n "
+                        + str(self.min_mismatches - 1)
+                        + "-l"
+                        + str(self.mismatch_region)
+                        + " "
+                        + file_probe_fasta_batch
+                        + " "
+                        + file_bowtie_batch
+                    )
+                else:
+                    command = (
+                        "bowtie transcriptome -f -a -v "
+                        + str(self.min_mismatches - 1)
+                        + " "
+                        + file_probe_fasta_batch
+                        + " "
+                        + file_bowtie_batch
+                    )
+
+                process = Popen(command, shell=True, cwd=self.dir_bowtie).wait()
+
+        index_exists = False
+        # Check if bowtie index exists
+        for file in os.listdir(self.dir_bowtie):
+            if re.search("^transcriptome*", file):
```

Comment:
should not be called transcriptome since it can be different regions -> it's the background DB instead